### PR TITLE
Give better error for greenfield API

### PR DESCRIPTION
**Pull request by NicolasDorier**
This attempt to remove generic errors with error carrying better information about what is going on.
There are probably still other places which need to be fixed.

### CODE REVIEW

**Review by Kukks — APPROVED** [review]
LGTM

File: BTCPayServer/Controllers/GreenField/InvoiceController.cs [hunk 1]
```diff
@@ -300,9 +304,19 @@ public async Task<IActionResult> ActivateInvoicePaymentMethod(string storeId, st
...
             }
-            return BadRequest();
+            ModelState.AddModelError(nameof(paymentMethod), "Invalid payment method");
+            return this.CreateValidationError(ModelState);
         }
-        
+
+        private IActionResult InvoiceNotFound()
+        {
+            return this.CreateAPIError(404, "invoice-not-found", "The invoice has not been found");
```
Comment (Zaxounette) [inline BTCPayServer/Controllers/GreenField/InvoiceController.cs:313]:
```suggestion
            return this.CreateAPIError(404, "invoice-not-found", "The invoice was not found");
```

File: BTCPayServer/Controllers/GreenField/InvoiceController.cs [hunk 2]
```diff
@@ -300,9 +304,19 @@ public async Task<IActionResult> ActivateInvoicePaymentMethod(string storeId, st
...
+        }
+        private IActionResult StoreNotFound()
+        {
+            return this.CreateAPIError(404, "store-not-found", "The store has not been found");
```
Comment (Zaxounette) [inline BTCPayServer/Controllers/GreenField/InvoiceController.cs:317]:
```suggestion
            return this.CreateAPIError(404, "store-not-found", "The store was not found");
```

**Review by Zaxounette — CHANGES_REQUESTED** [review]
Nitpicks on the wording or the errors.

**Review by dennisreimann — APPROVED** [review]

**Review by bolatovumar — APPROVED** [review]

**Pull request merged**